### PR TITLE
Rename hooks in plugin jest

### PR DESCRIPTION
**Pull request by royleejr**
## Description

Improvements #252 

Renamed project hooks and workspace hooks in `plugin-jest.ts` to keep consistency of our hook names with jest config names.

- Renamed `jestEnvironment` project hook to `jestTestEnvironment`
- Renamed `jestWatchIgnore` project hook to `jestWatchPathIgnorePatterns`.
- Renamed `jestTransforms` project hook to `jestTransform`.
- Renamed `jestModuleMapper` project hook to `jestModuleNameMapper`.
- Renamed `jestSetupEnv` project hook to `jestSetupFiles`.
- Renamed `jestSetupTest` project hook to `jestSetupFilesAfterEnv`.
- Renamed `jestSetupEnv` workspace hook to `jestSetupEnvironmentalFiles`.
- Renamed `jestSetupTests` workspace hook to `jestSetuptestsFiles`.

## Type of change

- [ ]  Patch: Bug (non-breaking change which fixes an issue)
- [ ]  Minor: New feature (non-breaking change which adds functionality)
- [ ]  Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by royleejr** [comment]
@BPScott I made the changes and updated the places that referenced those hooks and the checks pass! But I did notice one hook, `jestEnvironment` that I recently changed to`jestTestEnvironment`, the `jestEnvironment` being used in `plugin-build-library.ts` but wasn't references to the project hook that I renamed. Did you want me to rename that one as well for consistency or is okay to leave those as jestEnvironment?

File: packages/loom-plugin-jest/src/plugin-jest.ts [hunk 1]
```diff
@@ -27,24 +27,24 @@ type JestConfig = DeepReadonly<import('@jest/types').Config.InitialOptions>;
...
   readonly jestTestMatch: WaterfallHook<ReadonlyArray<string>>;
   readonly jestConfig: WaterfallHook<JestConfig>;
-  readonly jestWatchIgnore: WaterfallHook<ReadonlyArray<string>>;
+  readonly jestWatchPathIgnorePatterns: WaterfallHook<ReadonlyArray<string>>;
 }
 
 interface JestWorkspaceHooks {
-  readonly jestSetupEnv: WaterfallHook<ReadonlyArray<string>>;
-  readonly jestSetupTests: WaterfallHook<ReadonlyArray<string>>;
+  readonly jestSetupEnvironmentFiles: WaterfallHook<ReadonlyArray<string>>;
```
Comment (BPScott) [inline packages/loom-plugin-jest/src/plugin-jest.ts:46]: These two workspace hooks should match the names of the project hooks (and thus the jest options) too

jestSetupEnvironmentFiles should be named `jestSetupFiles`
jestSetupTestsFiles should be named `jestSetupFilesAfterEnv`

and

Comment (royleejr) [reply packages/loom-plugin-jest/src/plugin-jest.ts]: > These two workspace hooks should match the names of the project hooks (and thus the jest options) too
> 
> jestSetupEnvironmentFiles should be named `jestSetupFiles`
> jestSetupTestsFiles should be named `jestSetupFilesAfterEnv`

Just did this!

**Review by BPScott — COMMENTED** [review]
I pushed a tiny changelog adjustment, let me know what you think.

Looking great so far! I like the idea of keeping this consistent naming going all the way through to the public api options in `buildLibrary()`. Let's rename it's jestEnvironment option to `jestTestEnvironment` too. (this is a breaking change - remember to document it in that package's changelog)

**Comment by royleejr** [comment]
> I pushed a tiny changelog adjustment, let me know what you think.
> 
> Looking great so far! I like the idea of keeping this consistent naming going all the way through to the public api options in `buildLibrary()` / `jestConfig()` (plugin-build-library / plugin-jest-config). Let's rename their `jestEnvironment` options to `jestTestEnvironment` too. This is a breaking change - remember to document it in that package's changelog

Really liked the change! 

I also changed the options in `buildLibrary()` and `jestConfig()` and documented it in the changelog! Just pushed it up.

File: packages/loom-plugin-build-library/CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,10 @@ and adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ### Breaking Change
 
+- Renamed `jestEnvironment` option to `jestTestEnvironment` in buildLibrary and jestConfig functions. [[268](https://github.com/Shopify/loom/pull/268)]
```
Comment (BPScott) [inline packages/loom-plugin-build-library/CHANGELOG.md:12]: We've already got a heading called Breaking Change  - add this to the end of that list

**Pull request merged**